### ISSUE TITLE
Add garbage collection report at end of run

### ISSUE THREAD

**Issue by arielshaqed** [treeverse/lakeFS#3127]
When a GC job has finished, save the following data under the repository's storage namespace:
| Description                                                                                                                                                | Format  | Location                                                       |
|------------------------------------------------------------------------------------------------------------------------------------------------------------|---------|----------------------------------------------------------------|
| Basic information about the run, including:<br>- # of deleted objects<br>- The GC rules used for this jobs.                                                | JSON    | _lakefs/logs/gc/summary/dt=YYYY-MM-DD-HH-mm-SS/summary.json    |
| A set of parquet files describing all the ~deleted objects~ expired_addresses, including the columns:<br>- key (logical key in lakeFS)<br>- physical_address                   | Parquet | _lakefs/logs/gc/expired_addresses/dt=YYYY-MM-DD-HH-mm-SS/        |
| A list of all the commits participating in this GC job, including the columns:<br>- commit ID<br>- commit timestamp.<br>- whether it was expired or active | Parquet | _lakefs/logs/gc/summary/dt=YYYY-MM-DD-HH-mm-SS/commits.parquet |
| A list of all deleted objects                   | Parquet | _lakefs/logs/gc/deleted_objects/dt=YYYY-MM-DD-HH-mm-SS/deleted.parquet       |

Taken from the [product requirement for GC](#439)

Update:
When we implemented the report, we found the difference between expired_addresses, which are files that should be deleted, and the files that were actually deleted. We decided to report both, in case there will be a difference between them.

**Comment by talSofer** [comment]
Before we start working on this we should review [this](format https://github.com/treeverse/demo-lakefs/issues/439#issuecomment-1073244157) comment from Oz and discuss with @johnnyaug on the final path

**Comment by johnnyaug** [comment]
@talSofer @arielshaqed updated to new product requirement.

**Comment by lynnro314** [comment]
Closed with #3450. Added a new issue https://github.com/treeverse/lakeFS/issues/3527 to add the commit timestamp.
